### PR TITLE
[Doc] more details about sandboxing on macOS

### PR DESCRIPTION
**Pull request by almiliMSFT**

### CODE REVIEW

File: Documentation/Specs/Sandboxing.md [hunk 1]
```diff
@@ -22,11 +22,16 @@ In terms of performance, this implementation adds 1-5% of time overhead to runni
 Technical note: The top-level process initiating Detours calls must be a 64-bit process. Detours bootstrapping code is hard-coded to start from 64-bit, matching the requirements for large memory needs for the BuildXL engine for parsing and tracking large repos.
 
 ## MacOS Sandboxing
-Detouring is not a viable pattern on MacOS, so we use a kernel based implementation instead, but producing similar data and blocking capabilities as noted above for Windows.
+Interposing system calls (akin to Detouring on Windows) is possible on macOS, but comes with a major restriction: it is not applicable to "protected" system processes.  Another drawback of this approach is making sure that all relevant system calls are interposed, the list of which may be huge and not readily available.  Our sandbox for macOS avoids those restrictions by being implemented as a Darwin kernel extension, producing similar data and blocking capabilities as noted above for Windows.
```
Comment (erikma) [inline Documentation/Specs/Sandboxing.md]: macOS -> MacOS

Comment (almiliMSFT) [reply Documentation/Specs/Sandboxing.md]: For some reason, it's either macOS or Mac OS X, but not MacOS.  I couldn't find an article that explains why this is so; I'm sure @Kr1sso knows why.  The [macOS wiki page](https://en.wikipedia.org/wiki/MacOS), for example, consistently uses macOS.

Comment (Kr1sso) [reply Documentation/Specs/Sandboxing.md]: https://www.apple.com/macos/mojave/

Apple uses `macOS` consistently everywhere - so should we :)

Comment (erikma) [reply Documentation/Specs/Sandboxing.md]: See #47 for moe

**Pull request merged**